File: Lesson_5_Utkin/app/src/main/java/com/hito/nikolay/lesson_5_utkin/Adapter.java
package com.hito.nikolay.lesson_5_utkin;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class Adapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
    private ItemData[] mDataset;

    private CustomItemClickListener adapterClickListener;
    public Adapter(ItemData[] myDataset) {
        mDataset = myDataset;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View viewMain = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_recycler, parent, false);
        return new ViewHolderMain(viewMain, adapterClickListener);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        ViewHolderMain viewHolderMain = (ViewHolderMain) holder;
        viewHolderMain.bind(mDataset[position]);
    }

    @Override
    public int getItemCount() {
        return mDataset.length;
    }

    public interface CustomItemClickListener {
        void onItemClick(View view, int position);
    }
    public void setCustomItemClickListener(CustomItemClickListener customItemClickListener) {
        this.adapterClickListener = customItemClickListener;
    }
}

File: Lesson_8_Utkin/app/src/main/java/com/hito/nikolay/lesson_8_utkin/Adapter.kt
package com.hito.nikolay.lesson_8_utkin

import android.net.sip.SipAudioCall
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

class Adapter(
    private val bridgeList: ArrayList<Int>,
    private val listener: SipAudioCall.Listener
) : RecyclerView.Adapter<ViewHolderNote>() {

    interface Listener{
        fun onItemClick(bridge: Int)
    }
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolderNote {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getItemCount(): Int {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun onBindViewHolder(holder: ViewHolderNote, position: Int) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }
}